### PR TITLE
Adds falling hazard element, beware of falling tools, wear your hardhat, comically timed piano falling on the clown

### PR DESCRIPTION
**Pull request by Paxilmaniac**
## About The Pull Request

https://user-images.githubusercontent.com/82386923/199180691-6605c8cc-e8aa-490e-ab65-909d45d12ca0.mp4

Do note that the damage in this video is extremely exaggerated compared to what the normal value is.
## Why It's Good For The Game

All these signs about engineers needing to wear their hardhat, and for what? For the assistant dropping toolboxes onto them from above, that's what! Also allows people to do as god intended by allowing them to drop pianos on people.
## Changelog
:cl:
add: A variety of items, mainly tools, around the station might hurt if they fall on your head, remember to wear your hardhat and to avoid standing under large red X marks on the ground with a piano hanging above them.
/:cl:

### CODE REVIEW

**Comment by san7890** [comment]
I feel like it should at least force a rest if they get the weight of a piano on their head

**Comment by Paxilmaniac** [comment]
you have a point

**Comment by DAKKA-WAAAGH** [comment]
A full toolbox would annihilate anyone if dropped onto their head, regardless of the presence of a hardhat.

Still, spacemen are **BUILT DIFFERENT**. 💪

**Comment by Wallemations** [comment]
image

Me after getting my head indented by a falling toolbox


You should see if you can somehow implement the flattening function of vending machines for the piano so we can REALLY get the comedy rolling

**Comment by Paxilmaniac** [comment]
I have news for you, they do.

**Comment by Wallemations** [comment]
Oh my god I'm blind I was focusing too much on the guy who didn't fall
Amazing

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/datums/elements/falling_hazard.dm [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/datum/element/falling_hazard
+	element_flags = ELEMENT_BESPOKE | ELEMENT_DETACH
```
Comment (Mothblocks) [inline code/datums/elements/falling_hazard.dm:2]: You don't need ELEMENT_DETACH unless you are cleaning up stuff that isn't automatically cleaned up when the host is destroyed.

https://github.com/tgstation/tgstation/pull/70972

File: code/datums/elements/falling_hazard.dm [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+/datum/element/falling_hazard/Attach(datum/target, damage, wound_bonus, hardhat_safety, crushes)
+	. = ..()
+	if(!isatom(target))
+		return ELEMENT_INCOMPATIBLE
+
+	fall_damage = damage
+	fall_wound_bonus = wound_bonus
+	obeys_hardhats = hardhat_safety
+	crushes_people = crushes
```
Comment (Mothblocks) [inline code/datums/elements/falling_hazard.dm:24]: https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#when-passing-vars-through-new-or-initializes-arguments-use-srcvar

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Melbert beat a lot of me to it

**Comment by Serxule** [comment]
i love doing this on eris, cant wait for this to get merged

**Comment by Likteer** [comment]
Yes! YES! Jesus H. tap-dancing Christ! A multiZ piano interaction!

**Comment by Farquaar** [comment]
There should be a sound effect for items that fall on hard-hatted people. A good "tink" or "thunk" sound would make me smile

**Comment by Troxonater** [comment]
Great PR, I can't wait to make falling piano traps to troll security as clown.

File: code/datums/elements/falling_hazard.dm [hunk 3]
```diff
@@ -0,0 +1,80 @@
+/// An element that will make a target thing do damage to any mob that it falls on from a z-level above
+/datum/element/falling_hazard
+	element_flags = ELEMENT_BESPOKE
```
Comment (optimumtact) [inline code/datums/elements/falling_hazard.dm:3]: why use an element if it's just a component?

File: code/datums/elements/falling_hazard.dm [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	id_arg_index = 2
```
Comment (optimumtact) [inline code/datums/elements/falling_hazard.dm:4]: unless i'm crazy this means you're indexing by the damage which means every element with the same damage is going to have the same values for everything else based on the last element that gets created

Comment (optimumtact) [reply code/datums/elements/falling_hazard.dm:4]: arguable you could make a type index that would sort of make sense if you add this element to a lot of things

Comment (optimumtact) [reply code/datums/elements/falling_hazard.dm:4]: nevermind i'm wrong this is actually where it "Starts" hashing for uniqeness

File: code/datums/elements/falling_hazard.dm (hunk 3 above)
Comment (optimumtact) [reply code/datums/elements/falling_hazard.dm:3]: They said this is because the falling protection was an element too which I guess is fineeeeeee

**Pull request merged**